### DISCUSSION TITLE
RPi cannot be used to anything else besides WFB-ng?

### DISCUSSION THREAD

**Discussion by marceloroberti** [svpcom/wfb-ng#347] [Questions]
I'm trying to use WFB-ng with two RPi 4s, but I would like to use the drone onboard to capture images to be sent by WFB-ng. Is it possible or is the RPi just for creating WFB-ng communication?

Another question is whether the RPi Zero 2W can be used in WFB-ng? If not, what lightweight board could be used?

**Comment by svpcom** [comment]
Yes, it possible via ipv4 tunnel. See setup howto
Yes, Zero2W is equal to RPI3.

**Reply by marceloroberti** [reply]
Thank you. I'll check the howto again.

**Reply by marceloroberti** [reply]
Sorry, I couldn't figure it out.

I will describe what hardware I have and what I can do with them until now. Help me, please, if you can.

Drone side: Arduino + RPi4 + wifi adapter with RTL8812au
GS side: RPi4 + wifi adapter with RTL8812au + PC

I have this arduino collecting data from some sensors. It is sending data (mavlink and non-mavlink) through serial port of drone-RPi. Mavlink data I can see in my ground station PC, listening udp port 14550 with a python code or in QGC. Non-mavlink data I can't. I did not figure out how to configure IPv4 tunnel and where I can listen to this tunnel to get data on my ground station PC. I am using the prebuilt image for RPi3. Can you help me, please?

**Reply by svpcom** [reply]
1. enable default route via tunnel on the Drone (it is available as option in wifibroadcast.cfg. See master.cfg for reference)
2. add default route on the GS RPI to PC ip address. (but don't enable default route via tunnel in wifibroadcast.cfg)
3. On the PC do `route add 10.5.0.0/24 gw  address_of_your_gs_rpi`

after this you should able to ping 10.5.0.2 (drone RPI) or 10.5.0.1(gs RPI) from PC

**Reply by marceloroberti** [reply]
My PC is Windows and I'll use python to get this data. I can listen it in an udp port, as it's done with mavlink? This commands on step 3 are for linux, right?

**Reply by svpcom** [reply]
yes. but windows also have `route` command in terminal (cmd.exe). Or you can setup route in the windows gui

**Reply by marceloroberti** [reply]
I can just take GS RPi received data and send to the PC using an ethernet cable?

**Reply by marceloroberti** [reply]
I was able to meet my needs using the 'tunnel' message available in the Mavlink protocol. Thank you for your assistance @svpcom.
